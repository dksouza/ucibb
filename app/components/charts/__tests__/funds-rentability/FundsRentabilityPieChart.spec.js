import React from 'react';
import { shallow } from 'enzyme';
import FundsRentabilityPieChart from '../../funds-rentability';

let wrapper = {};
let props = {};

const title = 'Teste';
const data = [
  {
    fundName: 'MULTI DINAMICO V',
    value: 277.3,
    yieldValue: 0,
  },
  {
    fundName: 'MULTI DINAMICO V',
    value: 277.3,
    yieldValue: 0,
  },
  {
    fundName: 'MULTI DINAMICO V',
    value: 277.3,
    yieldValue: 0,
  },
  {
    fundName: 'MULTI DINAMICO V',
    value: 277.3,
    yieldValue: 0,
  },
  {
    fundName: 'MULTI DINAMICO V',
    value: 277.3,
    yieldValue: 0,
  },
  {
    fundName: 'MULTI DINAMICO V',
    value: 277.3,
    yieldValue: 0,
  },
  {
    fundName: 'MULTI DINAMICO V',
    value: 277.3,
    yieldValue: 0,
  },
  {
    fundName: 'MULTI DINAMICO V',
    value: 277.3,
    yieldValue: 0,
  },
  {
    fundName: 'MULTI DINAMICO V',
    value: 277.3,
    yieldValue: 0,
  },
  {
    fundName: 'MULTI DINAMICO V',
    value: 277.3,
    yieldValue: 1000,
  },
  {
    fundName: 'MULTI DINAMICO V',
    value: 277.3,
    yieldValue: 100,
  },
];

describe('FundsRentabilityPieChart', () => {
  beforeAll(() => {
    props = {
      data,
      title,
    };

    return props;
  });

  beforeEach(() => {
    wrapper = shallow(<FundsRentabilityPieChart {...props} />);
    return wrapper;
  });

  it('Shallow View', () => {
    expect(wrapper.find('View').exists()).toBe(true);
  });

  it('Shallow BBPieChart', () => {
    expect(wrapper.find('BBPieChart').exists()).toBe(true);
  });

  it('Shallow FundsRentabilityLegend - no title', () => {
    props = [...props.data, { title: '' }];
    expect(wrapper.find('FundsRentabilityLegend').exists()).toBe(true);
  });

  it('Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
