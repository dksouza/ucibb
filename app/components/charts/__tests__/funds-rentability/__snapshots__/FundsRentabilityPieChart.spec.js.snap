// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FundsRentabilityPieChart Snapshot 1`] = `
<View
  style={
    Object {
      "marginBottom": 32,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <BBPieChart
      data={Array []}
      innerLabel=""
      showCurrencySymbol={true}
      showInnerTotal={true}
    />
  </View>
  <FundsRentabilityLegend
    data={Array []}
  />
</View>
`;
